code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Offer.java

```diff
@@ -279,20 +278,26 @@ public OptimizeProposition getProposition() {
      * Dispatches an event for the Edge network extension to send an Experience Event to the Edge
      * network with the display interaction data for the given {@code OptimizeProposition} offer.
      *
-     * @see Offer#trackWithData(Map)
+     * @see OptimizeUtils#trackWithData(Map)
      */
     public void displayed() {
-        trackWithData(generateDisplayInteractionXdm());
+        if (propositionReference != null || propositionReference.get() != null) {
+            Map<String, Object> xdm = generateDisplayInteractionXdm();
+            propositionsRepository.trackDisplayInteraction(xdm);
+        }
     }
 
     /**
      * Dispatches an event for the Edge network extension to send an Experience Event to the Edge
      * network with the tap interaction data for the given {@code OptimizeProposition} offer.
      *
-     * @see Offer#trackWithData(Map)
+     * @see OptimizeUtils#trackWithData(Map)
      */
     public void tapped() {
-        trackWithData(generateTapInteractionXdm());
+        if (propositionReference != null || propositionReference.get() != null) {
```

Comment:
Same comment here. This check already exists in `generateTapInteractionXdm`. We should we able to directly call `OptimizeUtils.trackWithData(generateTapInteractionXdm())`

Reply:
done @spoorthipujariadobe